File: 0011.container-with-most-water/container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        if(height.size()==0) return 0;
        int area=0;
        int i=0,j=height.size()-1;
        while(i<j){
            int h=min(height[i],height[j]);
            area=max(area,h*(j-i));
            while(i<j && height[i]<=h) i++;
            while(i<j && height[j]<=h) j--;
        }
        return area;
    }
};

File: 0011.container-with-most-water/container-with-most-water.py
#不算水题，注意contain是矩形的，不是梯形
#http://www.jianshu.com/p/23e8e3e53abf
#http://blog.csdn.net/ljiabin/article/details/41673753
class Solution(object):
    def maxArea(self, height):
        """
        :type height: List[int]
        :rtype: int
        """
        i=0
        j=len(height)-1
        area=0
        while i!=j :
            area=max(area,min(height[i],height[j])*(j-i))
            if height[i]<height[j]:
                i+=1
            else :
                j-=1
        return area